driver-core/src/main/com/mongodb/connection/netty/NettySettings.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.connection.netty;
+
+import com.mongodb.connection.SslSettings;
+import com.mongodb.connection.TransportSettings;
+import com.mongodb.lang.Nullable;
+import io.netty.buffer.ByteBufAllocator;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.nio.NioEventLoopGroup;
+import io.netty.channel.socket.SocketChannel;
+import io.netty.channel.socket.nio.NioSocketChannel;
+import io.netty.handler.ssl.ReferenceCountedOpenSslClientContext;
+import io.netty.handler.ssl.SslContext;
+import io.netty.handler.ssl.SslContextBuilder;
+import io.netty.handler.ssl.SslProvider;
+
+import java.security.Security;
+
+import static com.mongodb.assertions.Assertions.isTrueArgument;
+import static com.mongodb.assertions.Assertions.notNull;
+
+/**
+ * A {@code TransportSettings} implementation for <a href="http://netty.io/">Netty</a>-based streams.
+ *
+ * @since 4.11
+ */
+public final class NettySettings extends TransportSettings {
+
+    private final EventLoopGroup eventLoopGroup;
+    private final Class<? extends SocketChannel> socketChannelClass;
+    private final ByteBufAllocator allocator;
+    @Nullable
+    private final SslContext sslContext;
+
+    /**
+     * Gets a builder for an instance of {@code NettyStreamFactoryFactory}.
+     * @return the builder
+     */
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    /**
+     * A builder for an instance of {@code NettyStreamFactoryFactory}.
+     */
+    public static final class Builder {
```

Comment:
Should this similarly extend a `TransportSettings.Builder` class that has no methods?  It doesn't seem useful.

Reply:
If we have such a hierarchy, `NettySettings.Builder.setCommonSomething` must return `NettySettings.Builder`, despite `TransportSettings.Builder.setCommonSomething` returning `TransportSettings.Builder`. It seems to me that we can introduce such a hierarchy later, when/if we actually need to have a `setCommonSomething` method. Unless you see a technical reason why postponing may be a problem, I prefer to postpone introducing the hierarchy now.